packages/backend-data/src/convert_js_resolvers.ts

```diff
@@ -18,17 +19,22 @@ const JS_PIPELINE_RESOLVER_HANDLER = './assets/js_resolver_handler.js';
  * It's required for defining a pipeline resolver. The only purpose it serves is returning the output of the last function in the pipeline back to the client.
  *
  * Customer-provided handlers are added as a Functions list in `pipelineConfig.functions`
+ *
+ * Add Amplify API ID and environment name to the context stash for use in the customer-provided handlers.
  */
-const defaultJsResolverAsset = (scope: Construct): Asset => {
+export const defaultJsResolverCode = (
+  amplifyApiId: string,
+  amplifyEnvironmentName: string
+): string => {
   const resolvedTemplatePath = resolve(
     fileURLToPath(import.meta.url),
     '../../lib',
     JS_PIPELINE_RESOLVER_HANDLER
   );
 
-  return new Asset(scope, 'default_js_resolver_handler_asset', {
-    path: resolveEntryPath(resolvedTemplatePath),
-  });
+  return readFileSync(resolvedTemplatePath, 'utf-8')
+    .replace('${amplifyApiId}', amplifyApiId)
+    .replace('${amplifyEnvironmentName}', amplifyEnvironmentName);
```

Comment:
For future proofing: If templates reference the match patterns more than once, this will fail. Consider a global regexp:
```suggestion
    .replace(new RegExp('\$\{amplifyApiId\}', 'g'), amplifyApiId)
    .replace(new RegExp('\$\{amplifyEnvironmentName\}', 'g'), amplifyEnvironmentName);
```

If you do adopt this change, please add tests to verify.